lib/evaluator/arenaWorker.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Web Worker entry point for the parallel arena evaluator.
+ *
+ * @module
+ */
+
+import { getEmbeddedReleaseWasm } from "./arenaWasm.embedded.ts";
+import type { ArenaWasmExports } from "./arenaEvaluator.ts";
+
+interface InitMessage {
+  type: "init";
+  memory: WebAssembly.Memory;
+  sab: SharedArrayBuffer;
+  workerId: number;
+}
+
+interface ConnectArenaMessage {
+  type: "connectArena";
+  arenaPointer: number;
+}
+
+let wasmExports: ArenaWasmExports | null = null;
+
+async function init(msg: InitMessage) {
+  const wasmBytes = getEmbeddedReleaseWasm().slice();
+  const module = await WebAssembly.compile(wasmBytes);
+  const instance = await WebAssembly.instantiate(module, {
+    env: {
+      memory: msg.memory,
+    },
+  });
+  wasmExports = instance.exports as unknown as ArenaWasmExports;
+
+  self.postMessage({ type: "ready", workerId: msg.workerId });
```

Comment:
research best practices if these should be human readable ascii high level js objects or raw machine code raw binary strings in Javascript, Google style, see this https://protobuf.dev/programming-guides/encoding/